### PR TITLE
Add validate number without region code

### LINKED ISSUE #64: Issues with Determining the ISO3166 Country Code given a phone number, validation
In our app I'm given a phone number. I don't know what country it is from, but we require the end user to enter the E164 +xxx country code.

With this library I cannot seem to turn a "+1 314 246 xxxx" phone number into the value "US" nor "+1 787 246 xxxx" into "PR"

Our customers are all over the world, and I don't want a "Country Code Picker" either. 

LibPhoneNumber can determine that the +1 314 246 xxxx is in the US, I'm not sure why this library does not expose it.

Help!

The big challenge here is that this library uses `isValidNumberForRegion()` as implemented in this library, in Java:

https://github.com/nashfive/phone_number/blob/d4ff262c4f91b2880f03a0d6072b5c7073772ad4/android/src/main/java/com/julienvignali/phone_number/PhoneNumberPlugin.java#L107

Now I have to know the region ahead of validation, but I don't know it, but it should be implied by the number, not requiring me to include the region. 

It would be nice to be able to have access to a different `.validate` function to match the Google LibPhoneNumber options, such as `.isPossibleNumber()` and `.isValidNumber()` rather than the very specific `.isValidNumberForRegion()`

Without this flexibility and access to the rest of the functionality of libphonenumber, I'm struggling to accomplish reliance on the value that LibPhoneNumber provides!

### PR DESCRIPTION
**Pull request by deandreamatias**
## Connection with issue(s)

Close #64 

## Solution description

If don't pass region code to validator, try validate a complete number

## Video demo

https://user-images.githubusercontent.com/21011641/182707477-f1a37f4b-345b-402d-8288-d7b6c43c4ba5.mp4

## To Do

- [x] Read [contributing guide](https://github.com/flutter-form-builder-ecosystem/.github/blob/main/CONTRIBUTING.md)
- [x] Check the original issue to confirm it is fully satisfied
- [x] Add solution description to help guide reviewers
- [ ] Add unit test to verify new or fixed behaviour
- [x] If apply, add documentation to code properties and package readme

### CODE REVIEW

**Comment by ooglek** [comment]
That looks great! If I do not pass a region code, still validate if the number is valid!

Thank you for digging into this. I can try to test it as well.

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ PhoneNumber{
 
 ### Validating
 
-Validating a phone number requires both the phone number string and the region country code.
+Validating a phone number requires the phone number string and optional the region country code.
 
 ```dart
 PhoneNumberUtil plugin = PhoneNumberUtil();
```
Comment (iamsahilsonawane) [inline README.md:57]: Will need to update (region as named param) the below examples

For e.g.: 
```diff
- bool isValid = await plugin.validate(springFieldUSASimpleNoRegion, region.code);
+ bool isValid = await plugin.validate(springFieldUSASimpleNoRegion, region: region.code);
```

**Review by iamsahilsonawane — CHANGES_REQUESTED** [review]

**Comment by iamsahilsonawane** [comment]
I understand that it's a WIP, still have added these comments as reference

**Comment by deandreamatias** [comment]
I tested in Android with random numbers and works fine

**Review by iamsahilsonawane — APPROVED** [review]
LGTM!

**Comment by ooglek** [comment]
When will this get pushed in a version bump? Thanks for fixing!

**Comment by deandreamatias** [comment]
When finish the readme refactor

**Comment by ooglek** [comment]
This breaks `format()` on IOS

https://github.com/flutter-form-builder-ecosystem/phone_number/blob/688e85155106a505385cab7c010fe750c64d6d7b/ios/Classes/SwiftPhoneNumberPlugin.swift#L26-L42

Passing `string` as `+5518988012364` and `region` as `US` and getting the error
PlatformException(InvalidArgument, The 'string' argument is missing., null, null)

**Comment by ooglek** [comment]
@deandreamatias Looks like the IOS Swift Code needs modification as well.

**Comment by deandreamatias** [comment]
Open a issue please

**Comment by ooglek** [comment]
Done, Issue #82

**Pull request merged**